core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -84,6 +91,26 @@ class BaseSnapshot implements Snapshot {
     this.manifestListLocation = manifestList;
   }
 
+  BaseSnapshot(long sequenceNumber,
+               long snapshotId,
+               Long parentId,
+               long timestampMillis,
+               String operation,
+               Map<String, String> summary,
+               Integer schemaId,
+               String manifestList) {
+    this.sequenceNumber = sequenceNumber;
+    this.snapshotId = snapshotId;
+    this.parentId = parentId;
+    this.timestampMillis = timestampMillis;
+    this.operation = operation;
+    this.summary = summary;
+    this.schemaId = schemaId;
+    this.manifestListLocation = manifestList;
+  }
+
+
+  @Deprecated // See note above
```

Comment:
To avoid the problem in SnapshotParser, I think we keep this un-deprecated.

Reply:
Removing the deprecation now.